miden-lib/build.rs

```diff
@@ -164,6 +172,61 @@ fn decrease_pow(line: io::Result<String>) -> io::Result<String> {
     Ok(line)
 }
 
+/// Generates `kernel_procs_v0.bin` and `kernel_proc_offsets.masm` files based on the kernel library
+fn generate_procedures_metadata(kernel: KernelLibrary) -> Result<()> {
+    let (_, module_info, _) = kernel.into_parts();
+
+    let to_exclude = BTreeSet::from_iter(["exec_kernel_proc"]);
+    let mut proc_digests_as_felts = vec![];
+    let mut generated_consts = String::new();
+    let mut generated_accessors = String::new();
+
+    module_info
+        .procedures()
+        .filter(|(_, proc_info)| !to_exclude.contains::<str>(proc_info.name.as_ref()))
+        .enumerate()
+        .for_each(|(offset, (_, proc_info))| {
+            proc_digests_as_felts.append(&mut proc_info.digest.as_elements().to_vec());
+
+            let name = proc_info.name.to_string();
+            let const_name = format!("{}_OFFSET", name.to_uppercase());
+
+            std::writeln!(&mut generated_consts, "const.{const_name}={offset}").unwrap();
+            std::writeln!(
+                &mut generated_accessors,
+                "\n#! Returns an offset of the `{name}` kernel procedure.\n\
+                #!\n\
+                #! Stack: []\n\
+                #! Output: [proc_offset]\n\
+                #!\n\
+                #! Where:\n\
+                #! - `proc_offset` is the offset of the `{name}` kernel procedure\n\
+                #! required to get the address where this procedure is stored.\n\
+                export.{name}_offset\n    push.{const_name}\nend"
+            )
+            .unwrap();
+        });
+
+    fs::write("kernel_procs_v0.bin", proc_digests_as_felts.to_bytes()).into_diagnostic()?;
```

Comment:
Instead of writing this as a binary file, it may be good to write it out as a `.rs` file. This way, we'd be able to visually see which procedures have changed.

Reply:
It was my first attempt actually. :) It will need to run rustfmt for formatting of the code, so I decided to make binary file. But it has its own disadvantages: we need to use `once_cell` for lazy initialization in `no-std` environment and of course binary is not human friendly. Thus, I agree with you, it will be better to step back and generate source code.

Reply:
We could also skip rustfmt formatting for this file.